src/internal/connector/exchange/data_collections.go

```diff
@@ -105,13 +105,13 @@ func parseMetadataCollections(
 
 				err := json.NewDecoder(item.ToReader()).Decode(&m)
 				if err != nil {
-					return nil, errors.New("decoding metadata json")
+					return nil, clues.New("decoding metadata json").WithClues(ctx)
 				}
 
 				switch item.UUID() {
 				case graph.PreviousPathFileName:
 					if _, ok := found[category]["path"]; ok {
-						return nil, errors.Errorf("multiple versions of %s path metadata", category)
+						return nil, clues.Wrap(clues.New(category.String()), "multiple versions of path metadata").WithClues(ctx)
```

Comment:
couple of questions for this and the call below:
* does context already contain the category?
* should we be adding category as a clue and not part of the error message? It seems more like related metadata to help with debugging rather than part of the root cause of the error

Reply:
> does context already contain the category?

Probably

> should we be adding category as a clue and not part of the error message? It seems more like related metadata to help with debugging rather than part of the root cause of the error

Depends on who you imagine consumes the error.  If this is only reviewed in logging, then it can stay as metadata.  If we imagine an end user (even if that's us) receiving it as, say, a cli output, the extra data can be really helpful.

To be honest, I don't think there's a clear rule for how to choose one or the other.  I'm a little arbitrary about the implementation myself.  But I also see this as a benign situation either way, so I don't put much emphasis a "correct" version.